src/AvoStorage.ts

```diff
@@ -5,53 +5,129 @@ export class AvoStorage {
   AsyncStorage: any | null = null;
 
   shouldLog: boolean;
-  initialized = false;
-  onInitFuncs: Array<() => void> = [];
+  storageInitialized = false;
+  itemsFromLastSessionLoaded = false;
+  useFallback = false;
+  fallbackStorage: any = {};
+  onStorageInitFuncs: Array<() => void> = [];
+  onItemsFromLastSessionLoadedFuncs: Array<() => void> = [];
 
   constructor(shouldLog: boolean) {
     this.shouldLog = shouldLog;
     if (!process.env.BROWSER) {
       this.reactNative = require("react-native");
       this.Platform = this.reactNative.Platform;
       this.AsyncStorage = this.reactNative.AsyncStorage;
-      
 
       if (this.Platform.OS === "android") {
+        // Android is the only platform that requires async loading of items from the prev session
+        this.initializeAndroid();
         this.AsyncStorage.getAllKeys().then((keys: Array<any>) =>
           this.AsyncStorage.multiGet(keys).then(
             (keyVals: Array<Array<string>>) => {
               keyVals.forEach((keyVal) => {
                 let key = keyVal[0];
                 this.data[key] = keyVal[1];
               });
-              this.initialized = true;
-              this.onInitFuncs.forEach((func) => {
-                func();
-              });
+              this.itemsLoadedAndroid();
             }
           )
         );
       } else {
-        this.initialized = true;
+        this.initializeAndItemsLoadedIos();
       }
     } else {
-      this.initialized = true;
+      // Verify localStorage
+      const uid = new Date().toISOString();
+      try {
+        window.localStorage.setItem(uid, uid);
+        if (window.localStorage.getItem(uid) === uid) {
+          window.localStorage.removeItem(uid);
+          this.initializeAndItemsLoadedWeb(true);
+        } else {
+          this.initializeAndItemsLoadedWeb(false);
+        }
+      } catch (error) {
+        this.initializeAndItemsLoadedWeb(false);
+      }
+    }
+  }
+
+  initializeAndroid() {
+    this.storageInitialized = true;
+    this.onStorageInitFuncs.forEach((func) => {
+      func();
+    });
+  }
+
+  itemsLoadedAndroid() {
+    this.itemsFromLastSessionLoaded = true;
+    this.onItemsFromLastSessionLoadedFuncs.forEach((func) => {
+      func();
+    });
+  }
+
+  initializeAndItemsLoadedIos() {
+    this.storageInitialized = true;
+    this.itemsFromLastSessionLoaded = true;
+    this.onStorageInitFuncs.forEach((func) => {
+      func();
+    });
+    this.onItemsFromLastSessionLoadedFuncs.forEach((func) => {
+      func();
+    });
+  }
+
+  initializeAndItemsLoadedWeb(localStorageAvailable: boolean) {
+    this.storageInitialized = true;
+    this.itemsFromLastSessionLoaded = true;
```

Comment:
is it ok to set `itemsFromLastSessionLoaded` to true for the first session or when localStorage is not available (no data available from last session)?

Reply:
I think so, in that case we would assume that there were no data in the last session